internals/cli/cli.go

```diff
@@ -141,15 +141,15 @@ func fixupArg(optName string) string {
 }
 
 type clientSetter interface {
-	setClient(*client.Client)
+	setClient(client.ClientGetter)
 }
 
-type clientMixin struct {
-	client *client.Client
+type ClientMixin struct {
+	client.ClientGetter
```

Comment:
For other reviewers: This is essentially the heart of the proposal. We need to change the CLI to accept an Client Interface, not only the Pebble Client structure. This will allow another pebble derivative to embed the Pebble Client struct and extend it with new options, and then initialise the Parser with the extended version.